actioncable/lib/action_cable/connection/stream.rb

```diff
@@ -29,8 +29,9 @@ def shutdown
       def write(data)
         return @rack_hijack_io.write(data) if @rack_hijack_io
         return @stream_send.call(data) if @stream_send
-      rescue EOFError
+      rescue EOFError, Errno::ECONNRESET
         @socket_object.client_gone
+        raise
```

Comment:
Bubble up to ClientSocket where it's rescued and emitted rather than swallowing it entirely.